src/renderer/shared/util/getBalances.js

```diff
@@ -37,10 +45,20 @@ export default async function getBalances({ net, address, tokens }) {
     const neoBalance = assets.NEO ? assets.NEO.balance.toString() : '0';
     const gasBalance = assets.GAS ? assets.GAS.balance.round(8).toString() : '0';
 
-    return {
+    const balances = {
       [NEO]: { scriptHash: NEO, balance: neoBalance, decimals: 0 },
       [GAS]: { scriptHash: GAS, balance: gasBalance, decimals: 8 }
     };
+
+    delete assets.NEO;
+    delete assets.GAS;
+
+    Object.keys(assets).forEach((tokenName) => {
+      const tokenData = getTokenData(assets, tokenName);
+      balances[tokenData.scriptHash] = tokenData;
+    });
+
+    return balances;
```

Comment:
There's another promise that comes before this one for fetching token balances.  If neoscan returns the token balances, then we can potentially drop the individual RPC calls for all token balances.  That comes at the tradeoff of trusting that neoscan is accurate, which Luciano (from CoZ) has said it is not.

Because it's not, we should probably prefer to keep the RPC request promise above for each token instead of getting this data from the API.

Reply:
fixed